CMakeLists.txt

```diff
@@ -184,7 +184,8 @@ if(BUILD_TESTING)
   set(CMAKE_CXX_STANDARD_REQUIRED ON)
   set(CMAKE_CXX_EXTENSIONS OFF) # prefer use of -std14 instead of -gnustd14
 
-  if(NOT TARGET gtest OR NOT TARGET gmock_main)
+  find_package(GTest REQUIRED)
```

Comment:
since you add `REQUIRED` -> if not found the cmake configure will fail where instead you would like to fallback to  "build googletest if not found in the system"